packages/wxt/src/core/builders/vite/plugins/devHtmlPrerender.ts

```diff
@@ -113,29 +113,22 @@ export function devHtmlPrerender(
     {
       name: 'wxt:virtualize-inline-scripts',
       apply: 'serve',
-      resolveId(id) {
-        // Resolve inline scripts
-        if (id.startsWith(virtualInlineScript)) {
+      resolveId: {
+        filter: {
+          id: [new RegExp(`^${virtualInlineScript}`)],
+        },
+        handler(id) {
           return '\0' + id;
-        }
-
-        // Ignore chunks during HTML file pre-rendering
-        if (id.startsWith('/chunks/')) {
-          return '\0noop';
-        }
+        },
       },
-      load(id) {
-        // Resolve virtualized inline scripts
-        if (id.startsWith(resolvedVirtualInlineScript)) {
-          // id="virtual:wxt-inline-script?<key>"
+      load: {
+        filter: {
+          id: [new RegExp(`^${resolvedVirtualInlineScript}`)],
+        },
+        handler(id) {
           const key = id.substring(id.indexOf('?') + 1);
           return inlineScriptContents[key];
-        }
-
-        // Ignore chunks during HTML file pre-rendering
-        if (id === '\0noop') {
-          return '';
-        }
```

Comment:
This needs to stay somehow. It dramatically speeds up dev startup speed by not bundling any of the JS code before writing the HTML files to disk.